server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -2510,6 +2508,12 @@ public Boolean doInTransaction(final TransactionStatus status) {
                 for (final VlanVO vlan : vlans) {
                     _vlanDao.remove(vlan.getId());
                 }
+                // we should actually
+//                VmwareDatacenterZoneMapVO vmwareZoneMapping = _vmwareDatacenterZoneMapDao.findByZoneId(zoneId)
+//                if (vmwareZoneMapping != null) {
+//                    _vmwareDatacenterZoneMapDao.removeByZoneId(vmwareZoneMapping.getId());
+//                }
+                // but we don't know about vmware/plugin/hypervisors at this point
```

Comment:
Is this needed?

Reply:
guess not, just informative
```suggestion
                // we should actually find the mapping and remove if it exists
                // but we don't know about vmware/plugin/hypervisors at this point
```